clippy_lints/src/escape.rs

```diff
@@ -81,7 +81,9 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
         let fn_def_id = cx.tcx.hir.local_def_id(node_id);
         let region_maps = &cx.tcx.region_maps(fn_def_id);
         {
-            let mut vis = ExprUseVisitor::new(&mut v, region_maps, &infcx);
+            let def_id = cx.tcx.hir.body_owner_def_id(body.id());
+            let param_env = cx.tcx.param_env(def_id);
```

Comment:
You can pass `fn_def_id`. Also, *never* use `borrowck_fake_infer_ctxt`. It's literally a hack for `borrowck`. Bad for everything else.